sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -34,6 +36,7 @@ import org.apache.spark.scheduler.{SparkListenerApplicationEnd, SparkListener}
  * `HiveThriftServer2` thrift server.
  */
 object HiveThriftServer2 extends Logging {
+  val SHUTDOWN_HOOK_PRIORITY: Int = 30
```

Comment:
I'd add a method in Utils.scala to add shutdown hooks, so that you can have a single constant and can avoid the ugly `new Runnable` syntax by using a Scala closure (`() => Unit`).

Not a big deal, but would make the patch a lot cleaner. :-)